loopin_test.go

```diff
@@ -87,10 +101,15 @@ func TestLoopInSuccess(t *testing.T) {
 	require.NotNil(t, state.HtlcTxHash)
 	require.Equal(t, cost, state.Cost)
 
-	// Expect register for htlc conf.
-	<-ctx.lnd.RegisterConfChannel
+	// Expect register for htlc conf (only one, the since the htlc is p2tr).
 	<-ctx.lnd.RegisterConfChannel
 
+	// If the swap is legacy, then we'll register two confirmation
+	// notifications.
+	if !s.isTaproot() {
+		<-ctx.lnd.RegisterConfChannel
```

Comment:
hmm..why do we need this extra conf? I think when `conf != nil` we'd exit the for loop in `waitForHtlcConf` right?